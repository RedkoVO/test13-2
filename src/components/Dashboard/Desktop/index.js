import React from 'react'
import PropTypes from 'prop-types'
import withStyles from '@material-ui/core/styles/withStyles'

import TransactionHistory from '../../../containers/TransactionHistory/AsyncTransactionHistory'
import BestAd from '../components/BestAd'
import Categories from '../components/Categories'
import Wallet from '../components/Wallet'
import Loyalty from '../components/Loyalty'
import Contacts from '../components/Contacts'
import Settings from '../components/Settings'

import styles from './styles'

const Dashboard = ({
  classes,
  bestAds,
  gameCategories,
  allBalance,
  isDisabledButton,
  onSubmit,
  onCopyKey,
  addressKey,
  handleChangeBalance,
  handleSubscribeCheckbox,
  isSubscribeCheckbox,
  curencySelectValue
}) => (
  <main>
    <div className={classes.root}>
      <Wallet
        onSubmit={onSubmit}
        onCopyKey={onCopyKey}
        addressKey={addressKey}
        isDisabledButton={isDisabledButton}
        allBalanceResult={allBalance && allBalance.result}
        handleChangeBalance={handleChangeBalance}
        curencySelectValue={curencySelectValue}
      />
      <TransactionHistory
        addressKey={addressKey}
        currency={curencySelectValue}
      />
      <Loyalty />
      <BestAd bestAds={bestAds} />
      <Categories categories={gameCategories} />
      <Contacts />
      <Settings
        handleSubscribeCheckbox={handleSubscribeCheckbox}
        isSubscribeCheckbox={isSubscribeCheckbox}
      />
    </div>
  </main>
)

Dashboard.propTypes = {
  classes: PropTypes.object,
  bestAds: PropTypes.array,
  gameCategories: PropTypes.array,
  allBalance: PropTypes.object,
  isDisabledButton: PropTypes.bool,
  onSubmit: PropTypes.func,
  onCopyKey: PropTypes.func,
  addressKey: PropTypes.string,
  handleChangeBalance: PropTypes.func,
  handleSubscribeCheckbox: PropTypes.func,
  isSubscribeCheckbox: PropTypes.bool,
  curencySelectValue: PropTypes.object
}

export default withStyles(styles)(Dashboard)
