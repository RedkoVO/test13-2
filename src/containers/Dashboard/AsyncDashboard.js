import Async from '../../components/App/components/Async'

export const AsyncDashboard = Async({
  loader: () => import(/* webpackChunkName: "dashboard_page" */ './index')
})

export default AsyncDashboard
