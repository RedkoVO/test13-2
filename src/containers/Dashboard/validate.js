const validate = values => {
  const errors = {}

  if (!values.addressKey) {
    errors.addressKey = 'error.required'
  } else if (values.addressKey.length !== 60) {
    errors.addressKey = 'error.length'
  }

  if (!values.amount) {
    errors.amount = 'error.required'
  } else if (!/^\d{1,19}(\.\d{1,8})?$/.test(values.amount)) {
    errors.amount = 'error.amount'
  }

  return errors
}

export default validate