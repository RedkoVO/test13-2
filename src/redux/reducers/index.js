import { combineReducers } from 'redux'
import { reducer as form } from 'redux-form'
import { APP_RESET } from '../actions/types'
import auth from './auth'
import balance from './balance'
import config from './config'
import history from './history'

const appReducer = combineReducers({
  form,
  auth,
  balance,
  config,
  history
})

export default (state, action) => {
  if (action.type === APP_RESET) {
    state = undefined
  }

  return appReducer(state, action)
}
