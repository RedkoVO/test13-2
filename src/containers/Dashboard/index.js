import compose from 'recompose/compose'
import { withProps, withHandlers, withState, lifecycle, pure } from 'recompose'
import { connect } from 'react-redux'
import { reduxForm, reset } from 'redux-form'
import isEqual from 'lodash/isEqual'
import BigNumber from 'bignumber.js'

import {
  getWork,
  getAllBalanceInfo,
  sendMoney
} from '../../redux/actions/balance'
import { getAllBalanceInfoSelector } from '../../selectors/balance'

import Crypto from '../../crypto/crypto'
import validate from './validate'
import { getBigNumberAmount } from '../../utils/math'

import AsyncDashboardDesktop from '../../components/Dashboard/Desktop/AsyncDashboardDesktop'

import BestShopLogo from '../../assets/images/shop.jpg'
import CSGOLogo from '../../assets/images/csgo.jpg'
import CasesLogo from '../../assets/images/cases.jpg'
import GameLogo from '../../assets/images/loto-game.png'
import CloverLogo from '../../assets/images/sim_game_icon_4clover.png'
import JungleImg2 from '../../assets/images/jungle_game2.png'

const tmpBestAdsCollection = [
  {
    id: 1,
    title: '',
    img: BestShopLogo,
    link: '/',
    titleLink: 'Stream of the day',
    nameLink: 'Blue cool Jelly'
  },
  {
    id: 2,
    title: '',
    img: CSGOLogo,
    link: '/skins',
    titleLink: 'Cases of the day',
    nameLink: 'Blue cool Jelly'
  },
  {
    id: 3,
    title: '',
    img: CasesLogo,
    link: '/cases',
    titleLink: 'Cases of the day',
    nameLink: 'Blue cool Jelly'
  }
]
const gameCategories = [
  {
    id: 1,
    game: [
      {
        id: 1,
        title: 'Most popular',
        name: '4Clover',
        img: CloverLogo,
        bundle: '4clover'
      },
      {
        id: 2,
        title: 'Most popular',
        name: 'Jungle Treasure',
        img: JungleImg2,
        bundle: 'jungle'
      },
      {
        id: 3,
        title: 'For children',
        name: 'Word games',
        img: GameLogo,
        bundle: '4clover'
      },
      {
        id: 4,
        title: 'For children',
        name: 'Funny games',
        img: GameLogo,
        bundle: '4clover'
      },
      {
        id: 5,
        title: 'For children',
        name: 'Word games',
        img: GameLogo,
        bundle: '4clover'
      },
      {
        id: 6,
        title: 'For children',
        name: 'Word games',
        img: GameLogo,
        bundle: '4clover'
      }
    ]
  },
  {
    id: 2,
    game: [
      {
        id: 1,
        title: 'Most popular',
        name: 'Loto games',
        img: GameLogo,
        bundle: '4clover'
      },
      {
        id: 2,
        title: 'Most popular',
        name: 'Dice games',
        img: GameLogo,
        bundle: '4clover'
      },
      {
        id: 3,
        title: 'For children',
        name: 'Word games',
        img: GameLogo,
        bundle: '4clover'
      },
      {
        id: 4,
        title: 'For children',
        name: 'Funny games',
        img: GameLogo,
        bundle: '4clover'
      },
      {
        id: 5,
        title: 'For children',
        name: 'Word games',
        img: GameLogo,
        bundle: '4clover'
      },
      {
        id: 6,
        title: 'For children',
        name: 'Word games',
        img: GameLogo,
        bundle: '4clover'
      }
    ]
  }
]

const FORM_NAME = 'sendMoney'

const mapStateToProps = state => ({
  allBalance: getAllBalanceInfoSelector(state)
})

export default compose(
  connect(mapStateToProps),
  reduxForm({
    form: FORM_NAME,
    validate
  }),
  withState('isDisabledButton', 'setDisabledButton', false),
  withState('curencySelectValue', 'setCurencySelectValue', {}),
  withState('textToCopy', 'setTextToCopy', false),
  withState('isSubscribeCheckbox', 'setisSubscribeCheckbox', true),
  withHandlers({
    handleChangeBalance: ({ setCurencySelectValue, allBalance }) => value => {
      setCurencySelectValue({
        currency: value,
        code: allBalance.result[value].code
      })
    },

    onCopyKey: ({ setTextToCopy, textToCopy }) => () => {
      setTextToCopy(!textToCopy)
    },

    handleSubscribeCheckbox: ({
      setisSubscribeCheckbox,
      isSubscribeCheckbox
    }) => () => {
      setisSubscribeCheckbox(!isSubscribeCheckbox)
    },

    onSubmit: ({
      handleSubmit,
      curencySelectValue,
      setDisabledButton,
      isDisabledButton,
      dispatch,
      allBalance
    }) =>
      handleSubmit(variables => {
        if (!isDisabledButton) {
          setDisabledButton(!isDisabledButton)

          dispatch(
            getWork({
              hash: allBalance.result[curencySelectValue.currency].lastBlock
            })
          )
            .then(res => {
              if (res.success) {
                const acc = {
                  publicKey: localStorage.getItem('publicKey'),
                  secretKey: localStorage.getItem('secretKey'),
                  address: localStorage.getItem('address')
                }
                const toAddress = variables.addressKey
                const amount = getBigNumberAmount(variables.amount)
                const work = res.result.work
                const currencyInfo = {
                  code: curencySelectValue.code,
                  currency: curencySelectValue.currency,
                  lastBlock:
                    allBalance.result[curencySelectValue.currency].lastBlock,
                  balance: new BigNumber(
                    allBalance.result[curencySelectValue.currency].balance
                  )
                }
                const getCryptoBlock = Crypto.sign.formSendBlock(
                  acc,
                  toAddress,
                  amount,
                  work,
                  currencyInfo
                )

                const data = {
                  block: getCryptoBlock
                }

                dispatch(sendMoney(data))
                  .then(res => {
                    if (res.result.hash) {
                      const getCryptoInfo = Crypto.account.accountFromSecret(
                        localStorage.getItem('secretKey')
                      )

                      if (localStorage.getItem('secretKey')) {
                        const data = {
                          address: getCryptoInfo.address
                        }
                        dispatch(getAllBalanceInfo(data))
                      }

                      dispatch(reset('sendMoney'))
                      setDisabledButton(false)
                    }
                  })
                  .catch(err => {
                    console.log('ERROR send money:', err)
                  })
              }
            })
            .catch(err => {
              console.log('ERROR get work:', err)
            })
        }
      })
  }),
  lifecycle({
    componentDidMount() {
      const { allBalance, setCurencySelectValue } = this.props
      if (!!allBalance) {
        setCurencySelectValue({
          code: allBalance.result['EUR'].code,
          currency: allBalance.result['EUR'].currency
        })
      }
    },
    componentDidUpdate(prevProps) {
      const {
        allBalance,
        setCurencySelectValue,
        curencySelectValue
      } = this.props
      if (!isEqual(prevProps.allBalance, allBalance)) {
        const curency = curencySelectValue.currency
          ? curencySelectValue.currency
          : 'EUR'

        if (!!allBalance.result[curency]) {
          setCurencySelectValue({
            code: allBalance.result[curency].code,
            currency: allBalance.result[curency].currency
          })
        }
      }
    }
  }),
  withProps(() => ({
    bestAds: tmpBestAdsCollection,
    gameCategories: gameCategories,
    addressKey: localStorage.getItem('address')
  })),
  pure
)(AsyncDashboardDesktop)
