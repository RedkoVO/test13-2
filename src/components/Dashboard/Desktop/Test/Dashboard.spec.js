import * as React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import { reduxForm } from 'redux-form'
import { createStore } from 'redux'
import { Provider } from 'react-redux'

import Dashboard from '../'

const dataBestAds = [
  {
    id: 1,
    title: 'Games',
    img: 'CloverLogo1',
    link: '/games',
    titleLink: 'Game of the day',
    nameLink: 'Smash Champs'
  },
  {
    id: 2,
    title: 'Shop',
    img: 'CloverLogo1',
    link: '/stream',
    titleLink: 'Product of the day',
    nameLink: 'Blue cool Jelly'
  }
]
const dataCategories = [
  {
    id: 1,
    game: [
      {
        id: 1,
        title: 'Most popular',
        name: '4Clover',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 2,
        title: 'Most popular',
        name: 'Dice games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 3,
        title: 'For children',
        name: 'Word games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 4,
        title: 'For children',
        name: 'Funny games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 5,
        title: 'For children',
        name: 'Word games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 6,
        title: 'For children',
        name: 'Word games',
        img: 'CloverLogo1',
        bundle: '4clover'
      }
    ]
  },
  {
    id: 2,
    game: [
      {
        id: 1,
        title: 'Most popular',
        name: 'Loto games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 2,
        title: 'Most popular',
        name: 'Dice games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 3,
        title: 'For children',
        name: 'Word games',
        img: 'CloverLogo1',
        bundle: '4clover'
      }
    ]
  }
]
const dataAllBalance = {
  result: [
    {
      balance: '0',
      currency: 'USD',
      lastBlock:
        '0000000000000000000000000000000000000000000000000000000000000000'
    },
    {
      balance: '1000000',
      currency: 'BTC',
      lastBlock:
        '0000000000000000000000000000000000000000000000000000000000000000'
    }
  ]
}

const Wrapper = reduxForm({ form: 'test' })(({ children }) => children)
const store = createStore(a => a, {})

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter keyLength={0}>
          <Wrapper>
            <Dashboard
              bestAds={dataBestAds}
              gameCategories={dataCategories}
              onSubmit={() => {}}
              onCopyKey={() => {}}
              isDisabledButton={false}
              allBalance={dataAllBalance}
              curencySelectValue={{ currency: 'DCB', code: 0 }}
              handleChangeBalance={() => {}}
              handleSubscribeCheckbox={() => {}}
              isSubscribeCheckbox={false}
            />
          </Wrapper>
        </MemoryRouter>
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
