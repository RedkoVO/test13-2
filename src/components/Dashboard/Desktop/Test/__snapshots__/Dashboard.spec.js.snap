// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main>
  <div
    className="Dashboard-root-1"
  >
    <div
      className="Wallet-root-2"
    >
      <div
        className="Wallet-wallet-3"
      >
        <div
          className="Wallet-walletTitle-4"
        >
          <p>
            My Address:
          </p>
        </div>
        <div
          className="Wallet-walletAddress-6"
        >
          <img
            alt=""
            className="Wallet-copyIcon-5"
            onClick={[Function]}
            src="copy.svg"
          />
        </div>
        <div
          className="MuiInputBase-root-34 MuiInput-root-21 MuiInput-underline-25 Wallet-selectCurency-7"
          onClick={[Function]}
        >
          <div
            className="MuiSelect-root-14"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="Wallet-selectField-9"
              id="select-curency"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              id="curencyDropdownWallet"
              name="curency"
              type="hidden"
              value="DCB"
            />
            <i
              className="Wallet-curencyIcon-10"
            />
          </div>
        </div>
        <ul
          className="Wallet-walletNav-12"
        >
          <li>
            <div
              className="Wallet-walletNavItem-13"
              onClick={[Function]}
            >
              DCB
            </div>
          </li>
          <li>
            <div
              className="Wallet-walletNavItem-13"
              onClick={[Function]}
            >
              USD
            </div>
          </li>
          <li>
            <div
              className="Wallet-walletNavItem-13"
              onClick={[Function]}
            >
              EUR
            </div>
          </li>
          <li>
            <div
              className="Wallet-walletNavItem-13"
              onClick={[Function]}
            >
              CPT
            </div>
          </li>
        </ul>
      </div>
      <div
        className="Action-root-55"
      >
        <div
          className="Action-actionsSend-60"
        >
          <form
            className="Action-form-58"
            onSubmit={[Function]}
          >
            <span
              className="Action-sendTitle-61"
            >
              Send now
            </span>
            <div
              className="Action-wrSendField-62"
            >
              <label
                htmlFor="addressKey"
              >
                Address:
              </label>
              <input
                className="Action-sendAddressField-63"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Put address you want to send the money"
                type="text"
                value=""
              />
            </div>
            <div
              className="Action-wrSendField-62"
            >
              <label
                htmlFor="amount"
              >
                Amount:
              </label>
              <input
                className="Action-sendAmmountField-64"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                placeholder="Set ammount"
                type="text"
                value=""
              />
            </div>
            <button
              className="Action-sendBtn-66"
              type="submit"
            >
              Send money
            </button>
          </form>
        </div>
        <div
          className="Action-actionsInvoice-68"
        >
          <span
            className="Action-invoiceTitle-69"
          >
            Invoice
          </span>
          <div
            className="Action-wrSendField-62"
          >
            <label
              htmlFor="invoice"
            >
              Email:
            </label>
            <input
              className="Action-invoiceAmmountField-65"
              name="invoice"
              placeholder="Set ammount you need to receive"
              type="text"
            />
          </div>
          <button
            className="Action-sendBtn-66"
            type="submit"
          >
            Send Invoice
          </button>
        </div>
      </div>
    </div>
    <section
      id="shops"
    >
      <div
        className="BestAd-bestTitle-73"
      >
        Shops
      </div>
      <div
        className="BestAd-root-70"
      >
        <div
          className="BestAd-bestItem-71"
        >
          <div
            className="BestAd-titleWrap-72"
          >
            <a
              className="BestAd-bestTitle-73"
              href="/games"
              onClick={[Function]}
            >
              Games
            </a>
          </div>
          <div
            className="BestAd-bestContent-74"
          >
            <a
              className="BestAd-bestImg-75"
              href="/games"
              onClick={[Function]}
            >
              <img
                alt=""
                className="BestAd-bestItemImg-76"
                src="CloverLogo1"
              />
            </a>
            <a
              className="BestAd-bestLink-77"
              href="/games"
              onClick={[Function]}
            >
              Game of the day
              : 
              <br />
              <span>
                Smash Champs
              </span>
            </a>
          </div>
        </div>
        <div
          className="BestAd-bestItem-71"
        >
          <div
            className="BestAd-titleWrap-72"
          >
            <a
              className="BestAd-bestTitle-73"
              href="/stream"
              onClick={[Function]}
            >
              Shop
            </a>
          </div>
          <div
            className="BestAd-bestContent-74"
          >
            <a
              className="BestAd-bestImg-75"
              href="/stream"
              onClick={[Function]}
            >
              <img
                alt=""
                className="BestAd-bestItemImg-76"
                src="CloverLogo1"
              />
            </a>
            <a
              className="BestAd-bestLink-77"
              href="/stream"
              onClick={[Function]}
            >
              Product of the day
              : 
              <br />
              <span>
                Blue cool Jelly
              </span>
            </a>
          </div>
        </div>
      </div>
    </section>
    <div
      className="Categories-root-78"
    >
      <div
        className="Categories-categories-79"
      >
        <div
          className="Categories-categTitleWrap-80"
        >
          <a
            className="Categories-categoriesTitle-81"
            href="/"
            onClick={[Function]}
          >
            Categories:
          </a>
        </div>
        <div
          className="CategorieItem-root-82"
        >
          <a
            className="CategorieItem-gameImg-83"
            href="/game/4clover"
            onClick={[Function]}
          >
            <img
              alt=""
              className="CategorieItem-gameImgLink-84"
              src="CloverLogo1"
            />
          </a>
          <a
            className="CategorieItem-gameLink-85"
            href="/game/4clover"
            onClick={[Function]}
          >
            Most popular
            : 
            <br />
             
            <span>
              4Clover
            </span>
          </a>
        </div>
        <div
          className="CategorieItem-root-82"
        >
          <a
            className="CategorieItem-gameImg-83"
            href="/game/4clover"
            onClick={[Function]}
          >
            <img
              alt=""
              className="CategorieItem-gameImgLink-84"
              src="CloverLogo1"
            />
          </a>
          <a
            className="CategorieItem-gameLink-85"
            href="/game/4clover"
            onClick={[Function]}
          >
            Most popular
            : 
            <br />
             
            <span>
              Dice games
            </span>
          </a>
        </div>
        <div
          className="CategorieItem-root-82"
        >
          <a
            className="CategorieItem-gameImg-83"
            href="/game/4clover"
            onClick={[Function]}
          >
            <img
              alt=""
              className="CategorieItem-gameImgLink-84"
              src="CloverLogo1"
            />
          </a>
          <a
            className="CategorieItem-gameLink-85"
            href="/game/4clover"
            onClick={[Function]}
          >
            For children
            : 
            <br />
             
            <span>
              Word games
            </span>
          </a>
        </div>
        <div
          className="CategorieItem-root-82"
        >
          <a
            className="CategorieItem-gameImg-83"
            href="/game/4clover"
            onClick={[Function]}
          >
            <img
              alt=""
              className="CategorieItem-gameImgLink-84"
              src="CloverLogo1"
            />
          </a>
          <a
            className="CategorieItem-gameLink-85"
            href="/game/4clover"
            onClick={[Function]}
          >
            For children
            : 
            <br />
             
            <span>
              Funny games
            </span>
          </a>
        </div>
        <div
          className="CategorieItem-root-82"
        >
          <a
            className="CategorieItem-gameImg-83"
            href="/game/4clover"
            onClick={[Function]}
          >
            <img
              alt=""
              className="CategorieItem-gameImgLink-84"
              src="CloverLogo1"
            />
          </a>
          <a
            className="CategorieItem-gameLink-85"
            href="/game/4clover"
            onClick={[Function]}
          >
            For children
            : 
            <br />
             
            <span>
              Word games
            </span>
          </a>
        </div>
        <div
          className="CategorieItem-root-82"
        >
          <a
            className="CategorieItem-gameImg-83"
            href="/game/4clover"
            onClick={[Function]}
          >
            <img
              alt=""
              className="CategorieItem-gameImgLink-84"
              src="CloverLogo1"
            />
          </a>
          <a
            className="CategorieItem-gameLink-85"
            href="/game/4clover"
            onClick={[Function]}
          >
            For children
            : 
            <br />
             
            <span>
              Word games
            </span>
          </a>
        </div>
      </div>
      <div
        className="Categories-categories-79"
      >
        <div
          className="Categories-categTitleWrap-80"
        >
          <a
            className="Categories-categoriesTitle-81"
            href="/"
            onClick={[Function]}
          >
            Categories:
          </a>
        </div>
        <div
          className="CategorieItem-root-82"
        >
          <a
            className="CategorieItem-gameImg-83"
            href="/game/4clover"
            onClick={[Function]}
          >
            <img
              alt=""
              className="CategorieItem-gameImgLink-84"
              src="CloverLogo1"
            />
          </a>
          <a
            className="CategorieItem-gameLink-85"
            href="/game/4clover"
            onClick={[Function]}
          >
            Most popular
            : 
            <br />
             
            <span>
              Loto games
            </span>
          </a>
        </div>
        <div
          className="CategorieItem-root-82"
        >
          <a
            className="CategorieItem-gameImg-83"
            href="/game/4clover"
            onClick={[Function]}
          >
            <img
              alt=""
              className="CategorieItem-gameImgLink-84"
              src="CloverLogo1"
            />
          </a>
          <a
            className="CategorieItem-gameLink-85"
            href="/game/4clover"
            onClick={[Function]}
          >
            Most popular
            : 
            <br />
             
            <span>
              Dice games
            </span>
          </a>
        </div>
        <div
          className="CategorieItem-root-82"
        >
          <a
            className="CategorieItem-gameImg-83"
            href="/game/4clover"
            onClick={[Function]}
          >
            <img
              alt=""
              className="CategorieItem-gameImgLink-84"
              src="CloverLogo1"
            />
          </a>
          <a
            className="CategorieItem-gameLink-85"
            href="/game/4clover"
            onClick={[Function]}
          >
            For children
            : 
            <br />
             
            <span>
              Word games
            </span>
          </a>
        </div>
      </div>
    </div>
    <section
      id="contacts"
    >
      <div
        className="Contacts-root-86"
      >
        <div
          className="Contacts-title-87"
        >
          Contacts
        </div>
        <div>
          <div
            className="Contacts-wrInfo-88"
          >
            <div
              className="Contacts-userPhoto-89"
            >
              <img
                alt=""
                src="user.jpg"
              />
            </div>
            <div>
              <div>
                Mike Grinch
              </div>
              <div>
                0x000000000000000000000000000000000000000000000000000000000
                <img
                  alt=""
                  className="Contacts-copyIcon-90"
                  src="copy.svg"
                />
              </div>
            </div>
          </div>
          <div
            className="Contacts-wrInfo-88"
          >
            <div
              className="Contacts-userPhoto-89"
            >
              <img
                alt=""
                src="user.jpg"
              />
            </div>
            <div>
              <div>
                Clarck Malkolm
              </div>
              <div>
                0x000000000000000000000000000000000000000000000000000000000
                <img
                  alt=""
                  className="Contacts-copyIcon-90"
                  src="copy.svg"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      id="settings"
    >
      <div
        className="Settings-root-91"
      >
        <div
          className="Settings-title-92"
        >
          Settings
        </div>
        <div
          className="Settings-themeTitle-93"
        >
          Theme
        </div>
        <div
          className="Settings-wrBtns-94"
        >
          <button
            className="Settings-btn-95"
          >
            Light
          </button>
          <button
            className="Settings-btn-95 Settings-disableBtn-96"
          >
            Dark
          </button>
        </div>
        <div>
          <span
            className="MuiButtonBase-root-113 MuiIconButton-root-107 MuiPrivateSwitchBase-root-103 MuiCheckbox-root-97 MuiCheckbox-colorPrimary-101"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label-112"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-116"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <input
                checked={false}
                className="MuiPrivateSwitchBase-input-106"
                data-indeterminate={false}
                onChange={[Function]}
                type="checkbox"
                value="checkedB"
              />
            </span>
            <span
              className="MuiTouchRipple-root-125"
            />
          </span>
          Subscribe for funds receiving
        </div>
      </div>
    </section>
  </div>
</main>
`;
